src/lib.rs

```diff
@@ -1389,6 +1465,122 @@ mod tests {
+        assert_eq!(response.code.unwrap(), 200);
+        assert_eq!(response.reason.unwrap(), "OK");
+        assert_eq!(response.headers.len(), 1);
+        assert_eq!(response.headers[0].name, "Line-Folded-Header");
+        assert_eq!(response.headers[0].value, &b"hello there"[..]);
+    }
+
+    static RESPONSE_WITH_OBSOLETE_LINE_FOLDING_AT_END: &'static [u8] =
+        b"HTTP/1.1 200 OK\r\nLine-Folded-Header: hello there\r\n   \r\n \r\n\r\n";
+
+    #[test]
+    fn test_forbid_response_with_obsolete_line_folding_at_end() {
+        let mut headers = [EMPTY_HEADER; 1];
+        let mut response = Response::new(&mut headers[..]);
+        let result = response.parse(RESPONSE_WITH_OBSOLETE_LINE_FOLDING_AT_END);
+
+        assert_eq!(result, Err(::Error::HeaderName));
+    }
+
+    #[test]
+    fn test_allow_response_with_obsolete_line_folding_at_end() {
+        let mut headers = [EMPTY_HEADER; 1];
+        let mut response = Response::new(&mut headers[..]);
+        let result = ::ParserConfig::default()
+            .allow_obsolete_multiline_headers_in_responses(true)
+            .parse_response(&mut response, RESPONSE_WITH_OBSOLETE_LINE_FOLDING_AT_END);
+
+        assert_eq!(result, Ok(Status::Complete(RESPONSE_WITH_OBSOLETE_LINE_FOLDING_AT_END.len())));
+        assert_eq!(response.version.unwrap(), 1);
+        assert_eq!(response.code.unwrap(), 200);
+        assert_eq!(response.reason.unwrap(), "OK");
+        assert_eq!(response.headers.len(), 1);
+        assert_eq!(response.headers[0].name, "Line-Folded-Header");
+        assert_eq!(response.headers[0].value, &b"hello there"[..]);
+    }
+
+    static RESPONSE_WITH_OBSOLETE_LINE_FOLDING_IN_MIDDLE: &'static [u8] =
+        b"HTTP/1.1 200 OK\r\nLine-Folded-Header: hello  \r\n \r\n there\r\n\r\n";
+
+    #[test]
+    fn test_forbid_response_with_obsolete_line_folding_in_middle() {
+        let mut headers = [EMPTY_HEADER; 1];
+        let mut response = Response::new(&mut headers[..]);
+        let result = response.parse(RESPONSE_WITH_OBSOLETE_LINE_FOLDING_IN_MIDDLE);
+
+        assert_eq!(result, Err(::Error::HeaderName));
+    }
+
+    #[test]
+    fn test_allow_response_with_obsolete_line_folding_in_middle() {
+        let mut headers = [EMPTY_HEADER; 1];
+        let mut response = Response::new(&mut headers[..]);
+        let result = ::ParserConfig::default()
+            .allow_obsolete_multiline_headers_in_responses(true)
+            .parse_response(&mut response, RESPONSE_WITH_OBSOLETE_LINE_FOLDING_IN_MIDDLE);
+
+        assert_eq!(result, Ok(Status::Complete(RESPONSE_WITH_OBSOLETE_LINE_FOLDING_IN_MIDDLE.len())));
+        assert_eq!(response.version.unwrap(), 1);
+        assert_eq!(response.code.unwrap(), 200);
+        assert_eq!(response.reason.unwrap(), "OK");
+        assert_eq!(response.headers.len(), 1);
+        assert_eq!(response.headers[0].name, "Line-Folded-Header");
+        assert_eq!(response.headers[0].value, &b"hello  \r\n \r\n there"[..]);
```

Comment:
I see, this was what I was wondering: how exactly does it get exposed. Am I right in assuming that this implies it will "simply" capture the text from the first line until the end of all obs-folded lines? So, if this were `hello: there\r\n mister\r\n baguette\r\n`, it would scoop it all as `b"there\r\n mister\r\n baguette"`?

That seems elegant. Just to consider what I'd thought of, I assumed this feature would instead use up an entry in the `&mut [Header]` slice for each line. If the line started with obs-folding, then the parse would just grab the name from the previous entry. So it'd look something like this:

```rust
assert_eq!(resp.headers[0].name, "hello");
assert_eq!(resp.headers[0].value, &b"there");
assert_eq!(resp.headers[1].name, "hello");
assert_eq!(resp.headers[1].value, &b"mister");
assert_eq!(resp.headers[2].name, "hello");
assert_eq!(resp.headers[2].value, &b"baguette");
```

Whatever we end up using, explaining that is what it does would be useful as a sentence/paragraph in the commit description.

Reply:
Line folds are equivalent to spaces, but multiple entries of a single header are equivalent to a single entry combined with commas, so we can't return multiple entries.

Reply:
Oh right.... gross...

Welp, then I think this is the best we can do. I think it'd be good to better explain, perhaps with an example, in the documentation for that builder method.